src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useMemo } from 'react';
+import { Todo } from '../../types/Todo';
+import { Status } from '../../types/Status';
+import { Filter } from '../Filter';
+
+interface Props {
+  todos: Todo[];
+  filterBy: Status;
+  setFilterBy: (newStatus: Status) => void;
+  onClearCompleted: () => void;
+}
+
+export const Footer: React.FC<Props> = React.memo(({
+  todos,
+  filterBy,
+  setFilterBy,
+  onClearCompleted,
+}) => {
+  const activeTodosLength = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length;
+  }, [todos]);
+
+  const hasCompleted = useMemo(() => {
+    return todos.some(todo => todo.completed);
+  }, [todos]);
```

Comment:
you don't need useMemo for these cases, they're too small and being used only in this component